<!-- Launches a Dataset Node and the ct_icp SLAM node -->
<launch>
  <arg name="dataset" required="true" doc="The name of the ct_icp dataset"/>
  <arg name="dataset_path" required="true" doc="Path to the root directory of the CT-ICP dataset"/>
  <arg name="sequence" required="true" doc="Sequence of the KITTI_360 CT-ICP dataset"/>
  <arg name="config" default="" doc="The path to the config"/>
  <arg name="frequency" default="10" doc="The frequency of frame acquisition"/>

  <!-- CT-ICP dataset -->
  <node name="ct_icp_dataset_node" pkg="ct_icp_odometry" type="ct_icp_dataset_node" output="screen">
      <param name="root_path" value="$(arg dataset_path)"/>
      <param name="sequence" value="$(arg dataset)_$(arg sequence)"/>
      <param name="dataset" value="$(arg dataset)"/>
      <param name="frequency_hz" value="$(arg frequency)"/>
  </node>

  <!-- CT-ICP Odometry -->
  <include file="$(find ct_icp_odometry)/launch/ct_icp_slam.launch">
       <arg name="config" value="$(arg config)"/>
       <arg name="pc_topic" value="ct_icp/pointcloud"/>
   </include>

</launch>
